File: cef-web/src/main/java/gov/epa/cef/web/service/EmissionsProcessService.java
package gov.epa.cef.web.service;

import gov.epa.cef.web.service.dto.EmissionsProcessDto;
import gov.epa.cef.web.service.dto.EmissionsProcessSaveDto;

import java.util.List;

import org.springframework.data.domain.Sort;

public interface EmissionsProcessService {

    /**
     * Create a new Emissions Process
     * @param dto
     * @return
     */
    EmissionsProcessDto create(EmissionsProcessSaveDto dto);

    /**
     * Update an Emissions Process
     * @param dto
     * @return
     */
    EmissionsProcessDto update(EmissionsProcessSaveDto dto);

    /**
     * Retrieve Emissions Process by its id
     * @param id
     * @return
     */
    EmissionsProcessDto retrieveById(Long id);

    /**
     * Retrieve Emissions Processes for a release point
     * @param pointId
     * @return
     */
    List<EmissionsProcessDto> retrieveForReleasePoint(Long pointId);

    /**
     * Retrieve Emissions Processes for an Emissions Unit
     * @param emissionsUnitId
     * @return
     */
    List<EmissionsProcessDto> retrieveForEmissionsUnit(Long emissionsUnitId);

    /**
     * Delete an Emissions Process for a given id
     * @param id
     */
    void delete(Long id);

}

File: cef-web/src/main/webapp/src/app/core/services/emissions-process.service.ts
import { Process } from 'src/app/shared/models/process';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class EmissionsProcessService {

  private baseUrl = 'api/emissionsProcess';  // URL to web api

  constructor(private http: HttpClient) { }

  create(process: Process): Observable<Process> {
    const url = `${this.baseUrl}`;
    return this.http.post<Process>(url, process);
  }

  update(process: Process): Observable<Process> {
    const url = `${this.baseUrl}/${process.id}`;
    return this.http.put<Process>(url, process);
  }

  /** GET specified release point from the server */
  retrieve(id: number): Observable<Process> {
    const url = `${this.baseUrl}/${id}`;
    return this.http.get<Process>(url);
  }

  retrievePrevious(id: number): Observable<Process> {
    const url = `${this.baseUrl}/${id}/previous`;
    return this.http.get<Process>(url);
  }

  retrieveForReleasePoint(releasePointId: number): Observable<Process[]> {
    const url = `${this.baseUrl}/releasePoint/${releasePointId}`;
    return this.http.get<Process[]>(url);
  }

  /**
   * GET all of the emissions processes for a specified emissions unit
   */
  retrieveForEmissionsUnit(emissionsUnitId: number): Observable<Process[]> {
    const url = `${this.baseUrl}/emissionsUnit/${emissionsUnitId}`;
    return this.http.get<Process[]>(url);
  }

    /** Delete specified emissions process from the database */
  delete(id: number): Observable<{}> {
    const url = `${this.baseUrl}/${id}`;
    return this.http.delete(url);
  }
}
